iptables/iptables.go

```diff
@@ -304,15 +305,21 @@ func Exists(table Table, chain string, rule ...string) bool {
 
 	// try -C
 	// if exit status is 0 then return true, the rule exists
-	if _, err := Raw(append([]string{
-		"-t", string(table), "-C", chain}, rule...)...); err == nil {
+	_, err := Raw(append([]string{"-t", string(table), "-C", chain}, rule...)...)
+	if err == nil {
 		return true
 	}
+	if !strings.Contains(err.Error(), "unknown option") {
+		return false
+	}
 
 	// parse "iptables -S" for the rule (this checks rules in a specific chain
 	// in a specific table)
-	ruleString := strings.Join(rule, " ")
-	ruleString = chain + " " + ruleString
+	return existsRaw(table, chain, rule...)
+}
+
+func existsRaw(table Table, chain string, rule ...string) bool {
+	ruleString := fmt.Sprintf("%s %s\n", chain, strings.Join(rule, " "))
```

Comment:
quite frankly, am not too comfortable depending on `\n` as a mechanism to guarantee the comparison and it could result in false-negatives depending on the position of the rule itself (and various other unrelated dependencies).